### DISCUSSION TITLE
RP2040 I2S RXTX mode

### DISCUSSION THREAD

**Discussion by thonk-luke** [pschatzmann/arduino-audio-tools#1040] [Q&A] [answered]
Hey all, after getting I2S output working with MCLK on the RP2040 I've now moved on to input, tried following the streams-i2s-serial example however I'm not getting any MCLK output when in RX mode. I've also tried the streams-i2s-i2s example where the MCLK is working properly, and there is data on the DOUT pin of my codec, however I am getting the error 
`[E] I2SRP2040.h : 48 - Unsupported mode: only TX_MODE is supported`

I have also tried the streams-i2s-i2s-2 example but am getting the error 
`error: 'class audio_tools::I2SConfigStd' has no member named 'port_no'
   29 |   
config_in.port_no = 0;`

Is this a hardware issue with the RP2040 or is it a case of just figuring out the right software setup? Thanks in advance

**Comment by pschatzmann** [comment, marked as answer]
The RP2040 does not support douplex mode where you could read and write at the same time¨
It will work with 2 I2SStream classes: Just comment out the port in the sketch since this is not needed&supported...

**Reply by thonk-luke** [reply]
Hey thanks for your reply, good to know! So this means I need two sets of clocks to my codec? I'm still not getting any MCLK output when using the I2S in RX mode

**Comment by pschatzmann** [comment]
I suggest to double check with the basic I2S API that comes with the RP2040.
https://arduino-pico.readthedocs.io/en/latest/i2s.html
